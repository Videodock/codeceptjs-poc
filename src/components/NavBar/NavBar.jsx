import React from 'react';
import * as PropTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import { makeStyles } from '@material-ui/styles';
import Button from '@material-ui/core/Button';
import AppbarTop from '../AppbarTop/AppbarTop';
import styles from './NavBar.styles';

const useStyles = makeStyles(styles);

export default function NavBar ({ navLinks, buttons }) {
  const classes = useStyles();

  const renderLinks = () => {
    return navLinks?.length > 0
      ? navLinks.map(({ name, to }) => (
        <NavLink key={name} className={classes.navLink} activeClassName={classes.navLinkActive} to={to}>
          {name}
        </NavLink>
      ))
      : null;
  };

  const renderButtons = () => {
    return buttons?.length > 0
      ? buttons.map(({ name, to }) => (
        <Button
          key={name}
          component={Link}
          to={to}
          variant="contained"
          color="secondary"
        >
          {name}
        </Button>
      ))
      : null;
  };

  return (
    <AppbarTop>
      <div>
        {renderLinks()}
        {renderButtons()}
      </div>
    </AppbarTop>
  );
}

NavBar.propTypes = {
  navLinks: PropTypes.array,
  buttons : PropTypes.array,
};
