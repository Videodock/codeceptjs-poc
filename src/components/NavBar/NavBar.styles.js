import { createStyles } from '@material-ui/styles';

export default ({ typography, palette, spacing }) => createStyles({
  navLink: {
    fontSize      : typography.pxToRem(15.8),
    letterSpacing : 0.5,
    textDecoration: 'none',
    color         : palette.text.darkGrey,
    marginRight   : spacing(4),
  },
  navLinkActive: {
    fontWeight: 600,
    color     : palette.secondary.main,
  },
});
