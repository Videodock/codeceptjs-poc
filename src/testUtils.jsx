import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
