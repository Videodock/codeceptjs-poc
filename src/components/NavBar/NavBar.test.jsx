import React from 'react';
import { createMemoryHistory } from "history";
import { Router } from "react-router";
import { fireEvent } from '@testing-library/react';
import { render } from '../../testUtils';
import NavBar from './NavBar';

describe('<NavBar />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render((
      <NavBar
        navLinks={[
          { name: 'Home', to: '/' },
        ]}
        buttons={[
          { name: 'Join campaign', to: 'campaign/e63fc3c5-f84e-4a64-9d5b-98a49dd4680c' },
        ]}
        renderDrawerContent={() => ''}
      />
    ));

    expect(container.firstChild).toMatchSnapshot();
  });

  test('displays navLink passed as prop', () => {
    const { getByText } = render((
      <NavBar
        navLinks={[
          { name: 'Home', to: '/' },
        ]}
        renderDrawerContent={() => ''}
      />
    ));

    expect(getByText('Home')).toBeTruthy();
  });

  test('click on navLink goes to correct route', () => {
    const history       = createMemoryHistory();
    const { getByText } = render((
      <Router history={history}>
        <NavBar
          navLinks={[
            { name: 'Home', to: '/' },
          ]}
          renderDrawerContent={() => ''}
        />
      </Router>
    ));

    fireEvent.click(getByText('Home'));
    expect(getByText('Home')).toBeTruthy();
    expect(history.location.pathname).toBe("/");
  });

  test('displays button passed as prop', () => {
    const { getByText } = render((
      <NavBar
        buttons={[
          { name: 'Join campaign', to: 'campaign/e63fc3c5-f84e-4a64-9d5b-98a49dd4680c' },
        ]}
        renderDrawerContent={() => ''}
      />
    ));

    expect(getByText('Join campaign')).toBeTruthy();
  });

  test('click on button goes to correct route', () => {
    const history       = createMemoryHistory();
    const { getByText } = render((
      <Router history={history}>
        <NavBar
          buttons={[
            { name: 'Join campaign', to: 'campaign/e63fc3c5-f84e-4a64-9d5b-98a49dd4680c' },
          ]}
          renderDrawerContent={() => ''}
        />
      </Router>
    ));

    fireEvent.click(getByText('Join campaign'));
    expect(getByText('Join campaign')).toBeTruthy();
    expect(history.location.pathname).toBe("/campaign/e63fc3c5-f84e-4a64-9d5b-98a49dd4680c");
  });
});
