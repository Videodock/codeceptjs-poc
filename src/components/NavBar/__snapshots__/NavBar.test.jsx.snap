// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-mobile-4"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-dense makeStyles-dense-8 makeStyles-mobile-15 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-hamburger-16"
      >
        <span
          aria-disabled="false"
          aria-label=""
          class="MuiButtonBase-root MuiIconButton-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </div>
      <a
        href="https://trompamusic.eu/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Trompa logo"
          class="makeStyles-logo-9"
          src="logo.png"
        />
      </a>
    </div>
  </header>
  <div
    class="PrivateSwipeArea-root-132 PrivateSwipeArea-anchorLeft-133"
    style="width: 20px;"
  />
</div>
`;
